### PR TITLE
Use Job states enum

### PR DESCRIPTION
**Pull request by kinow**
This is a small change with no associated Issue.

This `JobStates` enum is from the GScan deltas PR #542 . Last week on Friday afternoon I noticed the states in the User Profile page as strings.

Decided to take the enum from the GScan deltas PR as that will take a while to review. This gives us a chance to review part of that PR earlier, as well as already use it in th User Profile.

How it looks like:

image

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

File: src/components/cylc/tree/tree-nodes.js [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import TaskOutput from '@/model/TaskOutput'
+import TaskOutput from '@/model/TaskOutput.model'
```
Comment (kinow) [inline src/components/cylc/tree/tree-nodes.js:18]: Just for consistency with other files in the same folder

File: src/model/JobState.model.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Enumify } from 'enumify'
+
+/**
+ * Cylc valid job states.
+ */
+class JobState extends Enumify {
```
Comment (kinow) [inline src/model/JobState.model.js:23]: From #543

**Comment by kinow** [comment]
Firefox e2e tests fixed in #654

File: src/model/JobState.model.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  static SUBMIT_FAILED = new JobState('submit_failed')
+  static FAILED = new JobState('failed')
+  static RUNNING = new JobState('running')
+  static SUBMITTED = new JobState('submitted')
+  static READY = new JobState('ready')
+  static QUEUED = new JobState('queued')
+  static HELD = new JobState('held')
```
Comment (hjoliver) [inline src/model/JobState.model.js:30]: These used to be *task* (but not *job*) states, but are now not just indicator flags.
```suggestion
```

Definite list here: https://cylc.github.io/cylc-admin/proposal-state-names.html#taskjob-states

File: src/model/JobState.model.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Enumify } from 'enumify'
+
+/**
+ * Cylc valid job states.
+ */
+class JobState extends Enumify {
+  static SUBMIT_FAILED = new JobState('submit_failed')
```
Comment (hjoliver) [inline src/model/JobState.model.js:24]:
```suggestion
  static SUBMIT_FAILED = new JobState('submit-failed')
```

(The correct state name, and the css?, has a dash not an underscore)

**Review by hjoliver — CHANGES_REQUESTED** [review]

File: src/model/JobState.model.js (hunk 4 above)
Comment (kinow) [reply src/model/JobState.model.js]: Yup, nice catch. Fixed.

File: src/model/JobState.model.js (hunk 3 above)
Comment (kinow) [reply src/model/JobState.model.js]: Ha! Brilliant, we were using already the correct ones. Just updated the enum with the values from that page, and added a link in its jsdoc. Thanks!

**Review by oliver-sanders — APPROVED** [review]

**Review by hjoliver — APPROVED** [review]

**Pull request merged**